### PR TITLE
pkg/trace/{api,otlp,sampler}: enable priority sampling for OTLP Ingest

### PR DESCRIPTION
**Pull request by gbbr**
This change ensures that the priority sampler rates are used in OTLP as well, following user-set TPS rates, such as DD_APM_MAX_TPS, DD_APM_ERROR_TPS, etc. just like the Datadog tracing libraries.

### CODE REVIEW

File: pkg/trace/api/otlp_test.go [hunk 1]
```diff
@@ -87,14 +87,74 @@ var otlpTestTracesRequest = testutil.NewOTLPTracesRequest([]testutil.OTLPResourc
...
+	_, ok := o.rateKeys[serviceSignature{"a", "b"}]
+	require.False(t, ok)
+	o.rateKeys.Key("a", "b")
+	v, ok := o.rateKeys[serviceSignature{"a", "b"}]
+	require.True(t, ok)
+	require.Equal(t, "service:a,env:b", v)
+}
+
+func TestOTLPReceiverSamplingPriority(t *testing.T) {
+	mkspan := func(traceID uint64, service, env string) *pb.Span {
```
Comment (ajgajg1134) [inline pkg/trace/api/otlp_test.go:101]: `mkspan` doesn't actually use the `service` arg being provided here (although the tests seem to all pass as you only appear to use `a` for service name)

**Review by ajgajg1134 — CHANGES_REQUESTED** [review]
Overall looks pretty good! Just one small comment in the tests.
I think ideally there might also be some integration or system-level tests to verify this feature in perpetuity. It sounds like from chatting with @gbbr there is future work to support more integration tests for OTEL things, perhaps this can be included as a candidate there?

File: pkg/trace/api/otlp_test.go (hunk 1 above)
Comment (gbbr) [reply pkg/trace/api/otlp_test.go:101]: Haha, doh. Thanks for catching that.

**Review by ajgajg1134 — APPROVED** [review]
Overall looks good! We discussed some of the limitations in DMs, It might be worth including those in any documentation for this feature on the OTEL section of the datadog docs once this is ready for release :)

**Comment by gbbr** [comment]
Since this solution is not acceptable because some chunks might be dropped as opposed to others, resulting in incomplete distributed traces, I'm closing this PR.

**Pull request closed without merging**